### PR TITLE
merge in latest upstream / decrease image size / fix security issues

### PR DESCRIPTION
**Pull request by wochinge**
- merge in latest master from facebook
- use multistage docker image (image is just 2,7% of the original size,150MB vs 5.55 GiB(uncompressed))
- remove git installation due to https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-5260

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
+FROM haskell:8
+
+WORKDIR /duckling
+
+# Remove git since the installed version has a security leak and isn't required anyway
+RUN mkdir /log && apt purge -y git-man git
 
-CMD ["duckling-example-exe", "--no-access-log", "--no-error-log"]
+COPY --from=builder /root/.local/bin/duckling-example-exe .
+ENTRYPOINT [ "./duckling-example-exe" ]
```
Comment (wochinge) [inline Dockerfile]: these are my changes. The other ones are from the official Dockerfile

File: Dockerfile [hunk 2]
```diff
@@ -1,26 +1,32 @@
-FROM haskell:8
+FROM haskell:8 as builder
 
-RUN apt-get update -qq && \
-  apt-get install -qq -y libpcre3 libpcre3-dev build-essential --fix-missing --no-install-recommends && \
-  apt-get clean && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* 
+COPY . .
```
Comment (wochinge) [inline Dockerfile]: Actually I also replaced the `ADD` with a `COPY` since it's best practice

**Review by tmbo — APPROVED** [review]
looks good :+1: did you check that the API is still the same and that the docker container still works?

**Comment by wochinge** [comment]
Good point. The server started correctly, but the parsing itself didn't work. There was an open PR on the ducking repo for a multistage image (https://github.com/facebook/duckling/pull/341/) . Decreased the uncompressed size to a 140mb 🎉

**Pull request merged**